src/libs/Localize/index.js

```diff
@@ -91,6 +91,30 @@ function translateLocal(phrase, variables) {
     return translate(BaseLocaleListener.getPreferredLocale(), phrase, variables);
 }
 
+/**
+ *
+ * @param {Object|String} translation
+ * @returns {String}
+ */
+function translateIfNeeded(translation) {
+    if (_.isString(translation)) {
+        return translation;
+    }
+    const {key, params, transformer} = translation;
+    const variables = {};
+    if (params) {
+        _.keys(params).forEach((k) => {
+            // The value can be an object that need to translate
+            variables[k] = translateIfNeeded(params[k]);
```

Comment:
Is there a requirement for this now? Like objects inside params which need translation. 
I am just worried if we are bringing more complexity.

Reply:
Thanks for pointing out. I did miss that comment. Can we add adequate amount of documentation here? Since it will be hard to understand if we look back?